students/gerry_gabrisch/lesson03/slicing_lab.py

```diff
@@ -0,0 +1,146 @@
+def exchange_first_last(seq): 
+    '''Takes a sequence as input and places the last item first
+       the first item last, and leave all other remaining items 
+       in their original indexes.'''
+    #Get the last item in the sequence...
+    end = seq[-1]
+    #Get the first item in the sequence...
+    start = seq[0]
+    #Get everything in the sequence between the first and last elements...
+    seq = seq[1:-1]
+    #Lists have different ways of maninulating elements.  Check if it'
+    # a list and rebuild it.
+    if type(seq).__name__ == 'list':
+        seq.insert(0,end)
+        seq.append(start)
+        return seq 
+    #If the input was a string, identify it and add it together.
+    if type(seq).__name__ == 'str':
+        seq = end + seq + start
+        return seq 
+    #and because tuples are immutable concert it to a list, do the exchange, then cast
+    #the list back to a tuple...
+    if type(seq).__name__ == 'tuple':
+        seq = list(seq)
+        seq.insert(0,end)
+        seq.append(start)
+        return tuple(seq)         
```

Comment:
above is a bit more complex and does not follow slice syntax. this function should really be one liner and there should be no type checking.